app/assets/javascripts/discourse/app/controllers/user-activity-bookmarks.js

```diff
@@ -3,42 +3,31 @@ import { iconHTML } from "discourse-common/lib/icon-library";
 import Bookmark from "discourse/models/bookmark";
 import I18n from "I18n";
 import { Promise } from "rsvp";
-import EmberObject, { action } from "@ember/object";
+import EmberObject, { action, computed } from "@ember/object";
 import discourseComputed from "discourse-common/utils/decorators";
-import { notEmpty } from "@ember/object/computed";
+import { equal, notEmpty } from "@ember/object/computed";
+import { ajax } from "discourse/lib/ajax";
 
 export default Controller.extend({
   queryParams: ["q"],
+  q: null,
 
   application: controller(),
   user: controller(),
-
-  content: null,
   loading: false,
+  loadingMore: false,
   permissionDenied: false,
-  searchTerm: null,
-  q: null,
   inSearchMode: notEmpty("q"),
+  noContent: equal("model.bookmarks.length", 0),
 
-  loadItems() {
-    this.setProperties({
-      content: [],
-      loading: true,
-      permissionDenied: false,
-      searchTerm: this.q,
-    });
-
-    return this.model
-      .loadItems({ q: this.q })
-      .then((response) => this._processLoadResponse(response))
-      .catch(() => this._bookmarksListDenied())
-      .finally(() => {
-        this.setProperties({
-          loaded: true,
-          loading: false,
-        });
-      });
-  },
+  searchTerm: computed("q", {
```

Comment:
I don't understand why this get/set pattern is needed here on the computed property?

Reply:
We need this one-way binding: `q` → `searchTerm`. 

`q` is a search string from HTTP parameters, `searchTerm` is the property that's bound to the search box on the page. We need the one-way binding because we don't want `q` to be changed when a user is typing a search term. It would cause many requests to the server, we need to debounce them then. Also, it's a bit strange to see that the URL in the address bar is changing when you're typing something into the search box. We set `q` to the value of `searchTerm` only in the end, when a user presses <kbd>Enter</kbd>.

But when we do this update on <kbd>Enter</kbd>, one way binding stops working:
<img width="617" alt="Screenshot 2021-09-15 at 16 25 11" src="https://user-images.githubusercontent.com/1274517/133454655-beac8c31-cc4c-4b54-97cb-20d9db5ef490.png">

Because of this, sometimes after reentering the route with a new value of `q`, the search box on the page was still containing the previous search term. Defining setter resolves the problem. I would think of this pattern as about one-way binding with the ability to set the value when you need to (the value you're setting will persist until the property is recalculated).

Note, that you'll see this warning only if you implement a one-way binding like this:
```javascript
@computed("q")
searchTerm(q) {
   return q;
}),
```

If you use the [oneWay macros](https://api.emberjs.com/ember/release/functions/@ember%2Fobject%2Fcomputed/oneWay) behavior will be the same, but you won't see the warning. 

I initially was trying to implement it using the oneWay macros, had some hard time trying to figure out what's wrong :).

Reply:
Thanks for the thorough explanation, that makes sense!